zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -856,6 +856,23 @@ public void setPropertyAndRestart(String id, InterpreterOption option, Propertie
     }
   }
 
+  private boolean interpreterIsShared(InterpreterSetting setting) {
+    if (setting.getOption().isPerNoteProcess() == false &&
+      setting.getOption().isPerNoteSession() == false){
+      return true;
+    }
+    return false;
```

Comment:
Two things noted here:

1. Method body uses only `setting.getOption()` but not the `setting`, so it's better to just pass `setting.getOption()` as an argument to the function. This a general prinicple that is called ["Law of Demeter"](https://en.wikipedia.org/wiki/Law_of_Demeter) and it generally [helps to make code more testable](http://googletesting.blogspot.kr/2008/08/by-miko-hevery-so-you-decided-to.html).

2. Usually, returning just `true` or `false` after evaluating an exception means that the code can be simplified (given that 1 is implemented)

  ```java
return interpreterSetting.isPerNoteProcess() == false && interpreterSetting.isPerNoteSession() == false
  ```

  Which can be event further simplified to more readable and understandable:

 ```java
return interpreterSetting.isPerNoteProcess() || interpreterSetting.isPerNoteSession()
  ```

What do you think?